source/common/http/http1/codec_impl.cc

```diff
@@ -346,6 +346,8 @@ void ResponseEncoderImpl::encodeHeaders(const ResponseHeaderMap& headers, bool e
 
   // The contract is that client codecs must ensure that :status is present.
   ASSERT(headers.Status() != nullptr);
+  // encodeHeaders() is not supposed to throw any exceptions. This will crash on a release assert if
```

Comment:
I think it is possible that it could throw exception and exception and not crash when processing upstream response. The exception would have been caught by the dispatch() method of the client connection. Do we have an integration test where upstream responds with a garbage status?